### ISSUE TITLE
parse_integer should handle errors

### ISSUE THREAD

**Issue by darnir** [rockdaboot/wget2#25]
Currently the `parse_integer()` method uses the `atoi()` call to convert a string to integer. However, `atoi()` provides no error checking at all on the input string. Instead, we should use `strtol()` and check for errors.

As of now, it is possible to invoke wget2 as: `wget2 --max-threads=/dev/null example.com`. This causes Wget2 to be stuck in a deadlock of sorts since it spawns 0 downloader threads but is waiting for the queue to complete. 
We must also check for and reject when `--max-threads=0` since this makes no sense.

**Comment by rockdaboot** [comment]
There is
```
	// check for correct settings
	if (config.num_threads < 1)
		config.num_threads = 1;
```
You changed the code regarding max_threads/num_threads. Maybe max_threads should be checked here instead of num_threads !?

**Comment by rockdaboot** [comment]
I don't care to check everything and report it. Just at some point it must not lead to errors or unwanted behavior.

**Comment by darnir** [comment]
That was exactly why I'd asked for someone else to take a look. I thought I may have missed something, and I did!

While I don't generally support changing a user's explicit command without any information, here it seems like a sane idea to keep the application working. 

I was trying to work on reporting the conversion issue. But it will require a change in the prototype of the parse. So right now I'm debating if it is a good idea to do so.

**Comment by darnir** [comment]
Fixed in 6aa1e01.

About having more checking, I'll look into it later, but its not a necessity
